tests/stdlib/trstgen.nim

```diff
@@ -8,8 +8,37 @@ import ../../lib/packages/docutils/rstgen
 import ../../lib/packages/docutils/rst
 import unittest, strutils, strtabs
 
-proc toHtml(input: string): string =
-  rstToHtml(input, {roSupportMarkdown}, defaultConfig())
+proc toHtml(input: string,
+            rstOptions: RstParseOptions = {roSupportMarkdown},
+            expectError: string = "",
+            expectWarnings: seq[string] = @[]): string =
+  var warnings: seq[string]
+  var error: string
+  proc testMsgHandler(filename: string, line, col: int, msgkind: MsgKind,
+                      arg: string) =
+    let mc = msgkind.whichMsgClass
+    let a = $msgkind % arg
+    let message = "$1($2, $3) $4: $5" % [$filename, $line, $col, $mc, a]
```

Comment:
use `toLocation` std/private/miscdollars
rationale: DRY, and enables https://github.com/nim-lang/Nim/issues/690#issuecomment-706389479

Reply:
fixed though it's weird that `toLocation` omits column if it's 0. So I had to change starting column to 1, which would be worth doing anyway.

Reply:
i wrote it this way to avoid spurious `file.nim(1, 0)` messages when column is not generated in the 1st place (eg depending on nim's flags)